### PR TITLE
feat:add lists_data_key_format

### PR DESCRIPTION
**Pull request by zhouzhilong-commits**
## Summary by CodeRabbit

* **New Features**
  * Conditional Unix domain socket server support (enabled on Unix, no-op elsewhere).
  * Structured list storage keys with versioning, indexing and robust encode/decode handling.

* **Tests**
  * Added tests for list key encode/decode, edge cases, reserves and error conditions.

* **Chores**
  * Explicitly enabled additional async runtime features for networking dependency.
  * Adjusted formatting check to remove an unstable formatter flag.

### CODE REVIEW

File: src/storage/src/lists_data_key_format.rs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+use bytes::BytesMut;
+use std::mem;
+
+// Constants for fixed-length fields
+const RESERVE1_LEN: usize = 8;
+const RESERVE2_LEN: usize = 16;
+const U64_LEN: usize = 8;
+
+/*
+ * 用于 List 数据 key 的格式
```
Comment (AlexStocks) [inline src/storage/src/lists_data_key_format.rs]: no chinese comment

File: src/storage/src/lists_data_key_format.rs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            space: [0; 200],
+            reserve1: [0; 8],
+            key: key.to_vec(),
+            version,
+            index,
+            reserve2: [0; 16],
+        }
+    }
+
+    #[inline]
```
Comment (AlexStocks) [inline src/storage/src/lists_data_key_format.rs]: delete it

File: src/storage/src/lists_data_key_format.rs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+impl ParsedListsDataKey {
+    pub fn from_string(key: &str) -> Result<Self> {
+        Self::decode(key.as_bytes())
+    }
+
+    pub fn from_slice(key: &[u8]) -> Result<Self> {
+        Self::decode(key)
+    }
+
+    #[inline]
```
Comment (AlexStocks) [inline src/storage/src/lists_data_key_format.rs]: delete it

File: src/storage/src/lists_data_key_format.rs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        let pos = encoded_key_slice
+            .windows(ENCODED_KEY_DELIM_SIZE)
+            .position(|window| window == b"\x00\x00")
+            .map(|p| p + ENCODED_KEY_DELIM_SIZE)
+            .ok_or_else(|| crate::error::Error::InvalidFormat {
+                message: "Encoded key delimiter not found".to_string(),
+                location: snafu::location!(),
+            })?;
+
+        // 解码 user key
```
Comment (AlexStocks) [inline src/storage/src/lists_data_key_format.rs]: english comment

File: src/storage/src/lists_data_key_format.rs [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+const RESERVE2_LEN: usize = 16;
+const U64_LEN: usize = 8;
+
+/*
+ * 用于 List 数据 key 的格式
+ * | reserve1 | key | version | index | reserve2 |
+ * |    8B    |     |    8B   |   8B  |   16B    |
+ */
+pub struct ListsDataKey {
+    start: Option<Vec<u8>>,
```
Comment (guozhihao-224) [inline src/storage/src/lists_data_key_format.rs]: The C++ version uses 'start' to work with 'slice', which may not be necessary in the Rust version.

File: src/net/src/unix.rs [hunk 6]
```diff
@@ -1,21 +1,29 @@
-//  Copyright (c) 2017-present, arana-db Community.  All rights reserved.
+// Copyright (c) 2017-present, arana-db Community.  All rights reserved.
```
Comment (AlexStocks) [inline src/net/src/unix.rs]: https://github.com/arana-db/kiwi-rs/issues/80 根据这个格式改下 license 声明

File: src/storage/src/lists_data_key_format.rs [hunk 7]
```diff
@@ -0,0 +1,265 @@
+//  Copyright (c) 2017-present, arana-db Community.  All rights reserved.
```
Comment (AlexStocks) [inline src/storage/src/lists_data_key_format.rs]: https://github.com/arana-db/kiwi-rs/issues/80 同上，下面的也都改下吧

File: src/storage/src/lists_data_key_format.rs [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        Ok(())
+    }
+
+    #[test]
+    fn test_invalid_encoding() {
+        let invalid_data = b"invalid\x00\x02data";
+        let result = ParsedListsDataKey::from_slice(invalid_data);
+        assert!(matches!(result, Err(Error::InvalidFormat { .. })));
+    }
+}
```
[the remark this answers is not in the document]
Comment (AlexStocks) [reply src/storage/src/lists_data_key_format.rs]: 请根据 AI comment 改进测试

File: src/storage/src/coding.rs [hunk 9]
```diff
@@ -77,9 +77,10 @@ pub fn encode_fixed<T: FixedInt>(buf: *mut u8, value: T) {
 /// TODO: remove allow dead code
 #[allow(dead_code)]
 #[inline]
-pub fn decode_fixed<T: FixedInt>(buf: *mut u8) -> T {
+pub fn decode_fixed<T: FixedInt>(buf: *const u8) -> T {
     let size = T::byte_size();
-    let buffer: &mut [u8] = unsafe { std::slice::from_raw_parts_mut(buf, size) };
+    // SAFETY: Caller guarantees `buf` points to at least `size` bytes that are valid for reads.
+    let buffer: &[u8] = unsafe { std::slice::from_raw_parts(buf, size) };
```
Comment (flaneur2020) [inline src/storage/src/coding.rs]: it might be better to convert this buf to `Bytes` so that we can reduce the using of `unsafe` on decoding.

**Comment by flaneur2020** [comment]
there're some conflicts that requires to resolve

File: src/storage/src/coding.rs (hunk 9 above)
Comment (flaneur2020) [reply src/storage/src/coding.rs]: this pr could be considered as following the exisiting convention, we can consider merge this pr first, then having a seperated pr for the changes around `Bytes`.

File: src/storage/src/coding.rs [hunk 10]
```diff
@@ -77,9 +77,10 @@ pub fn encode_fixed<T: FixedInt>(buf: *mut u8, value: T) {
 /// TODO: remove allow dead code
 #[allow(dead_code)]
 #[inline]
-pub fn decode_fixed<T: FixedInt>(buf: *mut u8) -> T {
+pub fn decode_fixed<T: FixedInt>(buf: *const u8) -> T {
```
Comment (AlexStocks) [inline src/storage/src/coding.rs]: buf: *const u8 能否改为 buf: & [u8]？

Comment (AlexStocks) [reply src/storage/src/coding.rs]: 如果可以，其他类似地方也请一并改下
